### ISSUE TITLE
Clean DB

### ISSUE THREAD

**Issue by imidoriya** [peter-wangxu/persist-queue#126]
As I queue jobs and complete them, the database doesn't decrease in size that I can tell in my tests.  I just appears to grow and grow, saving the history.  Once a job is complete, I want it gone as my jobs are large in size and I don't want it to store any anything.  How do I keep the DB clean so it only contains pending jobs, not historical ones?

**Comment by peter-wangxu** [comment]
@imidoriya This feature is provided in master branch, give a try and kind me let know the result

Thanks
Peter

**Comment by imidoriya** [comment]
I wasn't able to see any shrinkage with my test.  I submitted several jobs where the db-wal increased in size up to 2.2MB and it just stayed there after all jobs were completed / acked and `clear_acked_data` and `shrink_disk_usage` were run.  So while I was able to test it fine, it didn't appear to do anything in my test.  I didn't see any errors reported.

Opening up the data.db, I still see 10 jobs which appear to have an Ack flag with a status of 5.  So it looks like the `clear_acked_data` may not be working as I expect it.  Looking the code, might it be the `LIMIT 1000 OFFSET {max_acked_length}`?  Would this mean that it's actually querying for 1001-2001?  I tried to set `q._MAX_ACKED_LENGTH = 0`, but didn't appear to work.

**Comment by imidoriya** [comment]
Actually, setting `q._MAX_ACKED_LENGTH = 0` did appear to fix `clear_acked_data` but `shrink_disk_usage` didn't decrease the db size, even though it's now empty.

**Comment by imidoriya** [comment]
Database is currently at 167MB for the data.db and 50MB for the data.db-wal.  The after clear_acked _data, the shrink_disk_usage didn't change the size.

**Comment by peter-wangxu** [comment]
@imidoriya sorry for the late response... 
the implementation of  `clear_acked_data` is somehow weird as it only marks 1000 items each time, so the followed   `shrink_disk_usage` may not shrink the file size dramatically.

I will follow up by a new pr to change the behavior of `clear_acked_data`

**Comment by peter-wangxu** [comment]
with the new change on my local, I can see the db size change, here is the sample code

```python
In [1]: from persistqueue import SQLiteAckQueue

In [2]: q = SQLiteAckQueue(path='data')

In [3]: s = "Database is currently at 167MB for the data.db and 50MB for the data.db-wal. The after clear_acked _data, the shrink_disk_usage didn't change the
   ...: size. %d"

In [4]: for x in range(200000):
   ...:     q.put(s % x)
   ...:     if x % 5000 == 0:
   ...:         print("done %d" % x)
   ...:
   
In [5]: for x in range(100000):
   ...:     item = q.get()
   ...:     q.ack(item)
   ...:     if x % 5000 == 0:
   ...:         print("ack %d" % x)

In [6]: ls -lh data

In [7]: q.clear_acked_data()

In [8]: q.shrink_disk_usage()

In [9]: del q
In [10]: ls -lh data/
```

**Comment by imidoriya** [comment]
Excellent - thank you :)

**Comment by imidoriya** [comment]
So this is odd - I had shrink_disk_usage disabled for a while and I just turned it back on.  I have the job running in a thread as documented here.  After it runs the shrink_disk_usage, the jobs no longer run.  It appears I can add new items to the queue, but maybe the loop gets closed as the jobs themselves no longer run.

**Comment by imidoriya** [comment]
I wonder if having multiple threads is preventing the vacuum from working properly and it just sort of hangs it.  Do you have any examples of perhaps more of an async wait_for_complete process - curious if that would work better than trying to use sqlite in a thread safe way.

**Comment by cnrmck** [comment]
Any update on this, I am also running into a persistently growing db and I need to run async?
